### ISSUE TITLE
torch.concat type hints fail for keyword argument

### ISSUE THREAD

**Issue by ikamensh** [pytorch/pytorch#82095]
### 🐛 Describe the bug

Using a keyword version of passing the axis causes type checkers (mypy, pyright) to reject working code:

```python 
import torch

a = (torch.randn([42]), torch.randn([42]))
torch.concat(a, 0)              # works fine!
torch.concat(a, axis=0)     # error: No overload variant matches argument types "Tuple[Tensor, Tensor]", "int" 
```

### Versions

PyTorch version: 1.12.0
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 11.6.6 (x86_64)
GCC version: Could not collect
Clang version: 13.0.0 (clang-1300.0.29.30)
CMake version: Could not collect
Libc version: N/A

Python version: 3.8.5 (default, Sep  4 2020, 02:22:02)  [Clang 10.0.0 ] (64-bit runtime)
Python platform: macOS-10.16-x86_64-i386-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] mypy==0.961
[pip3] mypy-extensions==0.4.3
[pip3] numpy==1.22.4
[pip3] torch==1.12.0
[conda] blas                      1.0                         mkl    defaults
[conda] cpuonly                   1.0                           0    pytorch
[conda] mkl                       2019.4                      233    defaults
[conda] mkl-service               2.3.0            py38hfbe908c_0    defaults
[conda] mkl_fft                   1.2.0            py38hc64f4ea_0    defaults
[conda] mkl_random                1.1.1            py38h959d312_0    defaults
[conda] numpy                     1.22.4                   pypi_0    pypi
[conda] torch                     1.12.0                   pypi_0    pypi

**Comment by ezyang** [comment]
Yeah, kwarg aliases are annoying to type. If someone wants to manually add in an extra alias for this case be my guess. There might be a way to programmatically do this too.
